### PR TITLE
tuplet: add version 2.0.0

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **tuplet/***

- add version 2.0.0
- use check_min_vs
- remove skip pylint
- use add_subdirectory in test_v1_package 

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by jwillikers — DISMISSED** [review]

File: recipes/tuplet/all/conanfile.py [hunk 1]
```diff
@@ -38,20 +37,22 @@ def package_id(self):
...
+            version = str(self.settings.compiler.version)
 
-        minimum_version = self._compilers_minimum_version.get(compiler, False)
-        if minimum_version and loose_lt_semver(version, minimum_version):
-            raise ConanInvalidConfiguration(
-                f"{self.name} {self.version} requires C++20, which your compiler ({compiler}-{version}) does not support")
+            minimum_version = self._compilers_minimum_version.get(compiler, False)
+            if minimum_version and loose_lt_semver(version, minimum_version):
+                raise ConanInvalidConfiguration(
+                    f"{self.name} {self.version} requires C++20, which your compiler ({compiler}-{version}) does not support")
```
Comment (SpaceIm) [inline recipes/tuplet/all/conanfile.py:55]: same comment than https://github.com/conan-io/conan-center-index/pull/14640#discussion_r1046366965, it's convoluted.

Comment (toge) [reply recipes/tuplet/all/conanfile.py]: reverted

File: recipes/tuplet/all/conanfile.py [hunk 2]
```diff
@@ -3,6 +3,7 @@
 from conan.tools.build import check_min_cppstd
 from conan.tools.files import copy, get
 from conan.tools.layout import basic_layout
+from conan.tools.microsoft import check_min_vs
```
Comment (SpaceIm) [inline recipes/tuplet/all/conanfile.py:6]:
```suggestion
```

File: recipes/tuplet/all/conanfile.py [hunk 3]
```diff
@@ -26,8 +27,6 @@ def _min_cppstd(self):
     def _compilers_minimum_version(self):
         return {
             "gcc": "11",
```
Comment (SpaceIm) [inline recipes/tuplet/all/conanfile.py:29]:
```suggestion
            "gcc": "11",
            "Visual Studio": "17",
            "msvc": "193",
```

File: recipes/tuplet/all/conanfile.py [hunk 4]
```diff
@@ -38,6 +37,7 @@ def package_id(self):
     def validate(self):
         if self.settings.compiler.get_safe("cppstd"):
             check_min_cppstd(self, self._min_cppstd)
+        check_min_vs(self, 193)
```
Comment (SpaceIm) [inline recipes/tuplet/all/conanfile.py:40]:
```suggestion
```

**Review by SpaceIm — CHANGES_REQUESTED** [review]
I meant revert completly check_min_vs

**Review by uilianries — APPROVED** [review]
LGTM

**Review by jwillikers — APPROVED** [review]

**Pull request merged**